### DISCUSSION TITLE
[Simulation question]:Regarding whether custom low-level code design is required for cell modeling.

### DISCUSSION THREAD

**Discussion by aoliaodaxiaoguaishou** [topas-nbio/TOPAS-nBio#30] [Simulation-question] [answered]
### Simulation question

Hello, I am working on a cell model that includes an elliptical cell, cytoplasm, and cell membrane. However, after visualization, only the elliptical cell is visible, meaning that the cytoplasm and cell membrane are not showing up. Additionally, after modifying the cytoplasm code, the terminal gave the following error message:
##Topas is exiting due to a serious error for material name: Cytoplasm
This material has no density defined##
[My_cell_test.txt](https://github.com/user-attachments/files/16735855/My_cell_test.txt)

Now I have the following questions:
Is the part of the code on the left side of the equals sign not just a simple declaration, but is it interconnected with the code on the right side, and both are compiled together by the underlying code? Therefore, changes to the left side code cannot be made arbitrarily?
For visualizing the cell membrane and cytoplasm, do I need to write or modify the underlying code to achieve a visualization effect that includes the elliptical cell, cytoplasm, and cell membrane?

### TOPAS-nBio category

Geometry

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment, marked as answer]
Hi there,
The left part of the equal sign is the TOPAS-nBio parameter. This must be available in TOPAS-nBio to make changes in the simulation's configuration. In addition, it can be defined by the user as a new auxiliary parameter to simplify the simulation configuration. E.g., 

    s:Ge/MyCell/Material  ...  <--- Existing TOPAS parameter
    s:Ge/MyUserDefinedNewMaterial ... <--- New user-defined parameter, it doesn't nothing by itself
    # But as follows will benefit or help to define existing parameters:
    s:Ge/MyCell/Material =   Ge/MyUserDefinedNewMaterial   <--- user-defined parameter to set a TOPAS parameter

From your text file, the material "Cytoplasm" has not been defined elsewhere, thus the error is shown. The configuration of the elliptical cell is a bit different than yours. Please see the example TOPAS-nBio/examples/geometry/cells/EllipsoidCell.txt, and modify the parameter values according to your needs. 
Cheers!

**Reply by aoliaodaxiaoguaishou** [reply]
Thank you very much for your reply. I have understood and solved this problem! Your support has given me great confidence and help. Thank you again!
